string battery_estimate
string controller
